src/cmlibs/utils/zinc/group.py

```diff
@@ -24,15 +85,27 @@ def group_add_group_elements(group: FieldGroup, other_group: FieldGroup, highest
                 other_mesh_group = other_group.getMeshGroup(mesh)
                 if other_mesh_group.isValid() and (other_mesh_group.getSize() > 0):
                     mesh_group = group.getOrCreateMeshGroup(mesh)
```

Comment:
Sorry I'm a little confused about this comment. Do you mean that if we are doing a removal and `mesh_group` is empty then we should skip performing the operations for that iteration? Why should I use `getMeshGroup` instead of `getOrCreateMeshGroup`?

Reply:
Because the MeshGroups and NodesetGroups are created on demand, we don't want to make empty ones for no good reason in these cases:

1.  When removing from a group; if there is no current mesh/nodeset group we know we can't remove anything from it and we don't want to make an empty such group, hence use 'get' and only remove if isValid()
2. When using objects contained in a source group we should never modify it, so always use 'get' and check it isValid() before using. If it isn't valid, there is nothing to act on.